### PR TITLE
Issue#2210 : Unit Tests for SmtpDataStuffing (2)

### PR DESCRIPTION
**Pull request by shonramamurthy**
Implemented feedback from #2228 
Let me know if more tests are needed or if I'm missing something.

### CODE REVIEW

File: k9mail-library/src/test/java/com/fsck/k9/mail/filter/SmtpDataStuffingTest.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+import org.junit.Test;
+
+import static junit.framework.Assert.assertEquals;
+
+public class SmtpDataStuffingTest {
+    private String expectedMessageAfterStuffing;
+    private String actualMessageAfterStuffing;
+
+    @Before
+    public void setUp(){
```
Comment (Valodim) [inline k9mail-library/src/test/java/com/fsck/k9/mail/filter/SmtpDataStuffingTest.java]: code style

File: k9mail-library/src/test/java/com/fsck/k9/mail/filter/SmtpDataStuffingTest.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        byte[] data = ByteString.encodeUtf8("Hello dot\r\n.").toByteArray();
+
+        expectedMessageAfterStuffing = "Hello dot\r\n..";
+        try {
+            smtpDataStuffingStream.write(data);
+        }
+        finally {
+            smtpDataStuffingStream.close();
+            byteArrayOutputStream.close();
+        }
```
Comment (Valodim) [inline k9mail-library/src/test/java/com/fsck/k9/mail/filter/SmtpDataStuffingTest.java]: if test code breaks it's okay to break uncleanly, no need for the try/finally here. (also, code style on the } )

File: k9mail-library/src/test/java/com/fsck/k9/mail/filter/SmtpDataStuffingTest.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+import okio.ByteString;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import static junit.framework.Assert.assertEquals;
+
+public class SmtpDataStuffingTest {
+    private String expectedMessageAfterStuffing;
+    private String actualMessageAfterStuffing;
```
Comment (Valodim) [inline k9mail-library/src/test/java/com/fsck/k9/mail/filter/SmtpDataStuffingTest.java]: is there a reason you put these into fields?

File: k9mail-library/src/test/java/com/fsck/k9/mail/filter/SmtpDataStuffingTest.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+    @Before
+    public void setUp(){
+        expectedMessageAfterStuffing = "";
+        actualMessageAfterStuffing = "";
+    }
+
+    @Test
+    public void smtpDotStuffing_OnlyStuffingDotTest() throws IOException {
```
Comment (Valodim) [inline k9mail-library/src/test/java/com/fsck/k9/mail/filter/SmtpDataStuffingTest.java]: snakeCase_isLowercase_afterUnderscore

File: k9mail-library/src/test/java/com/fsck/k9/mail/filter/SmtpDataStuffingTest.java (hunk 3 above)
Comment (shonramamurthy) [inline k9mail-library/src/test/java/com/fsck/k9/mail/filter/SmtpDataStuffingTest.java]: No particular reason; now that you mention it looks superfluous. Should I move them inside individual methods?

File: k9mail-library/src/test/java/com/fsck/k9/mail/filter/SmtpDataStuffingTest.java (hunk 4 above)
Comment (shonramamurthy) [inline k9mail-library/src/test/java/com/fsck/k9/mail/filter/SmtpDataStuffingTest.java]: My bad, sure will fix this.

File: k9mail-library/src/test/java/com/fsck/k9/mail/filter/SmtpDataStuffingTest.java (hunk 1 above)
Comment (shonramamurthy) [inline k9mail-library/src/test/java/com/fsck/k9/mail/filter/SmtpDataStuffingTest.java]: Sure, I'll add a space (of course, if I move the fields inside the methods this wouldn't be needed).

File: k9mail-library/src/test/java/com/fsck/k9/mail/filter/SmtpDataStuffingTest.java (hunk 2 above)
Comment (shonramamurthy) [inline k9mail-library/src/test/java/com/fsck/k9/mail/filter/SmtpDataStuffingTest.java]: Sure, I'll do the same! Do you mean a line break after line 36 here (for code style)?

**Comment by shonramamurthy** [comment]
I've updated changes from the review for #2210 . Let me know if anything else is required.

**Comment by cketti** [comment]
Thanks!

I cleaned up the tests a bit and merged manually. See https://github.com/k9mail/k-9/commit/47b86f6146f5e95f1027e6e129e1febf794ff5b8

**Pull request closed without merging**